app/models/notifications/mentioned_in_post.rb

```diff
@@ -0,0 +1,28 @@
+module Notifications
+  class MentionedInPost < Notification
+    include Notifications::Mentioned
+
+    def mail_job
+      Workers::Mail::Mentioned
+    end
+
+    def popup_translation_key
+      "notifications.mentioned"
+    end
+
+    def deleted_translation_key
+      "notifications.mentioned_deleted"
+    end
+
+    def self.filter_mentions(mentions, mentionable, recipient_user_ids)
+      return mentions if mentionable.public
+      mentions.where(person: Person.where(owner_id: recipient_user_ids).ids)
+    end
+
+    def self.people_allowed_to_be_notified(mentions_container)
+      super.tap do |ppl|
```

Comment:
hmm, after thinking about this again: you don't even need this here, because `filter_mentions` does already the correct thing for mentions on posts. And for mentions on comments it is probably better to filter allowed mentions in `filter_mentions` too, so we don't have to filter twice.